cobra/flux_analysis/loopless.py

```diff
@@ -200,57 +201,52 @@ def loopless_fva_iter(model, reaction, all_fluxes=False, zero_cutoff=1e-9):
-    model.solver.objective.set_linear_coefficients(
-        {model.variables.fva_old_objective: 1,
-         reaction.forward_variable: 0, reaction.reverse_variable: 0})
-
-    loopless = loopless_solution(model)
-
-    # If the previous optimum is maintained in the loopless solution it was
-    # loopless and we are done
-    if abs(loopless[reaction.id] - current) < zero_cutoff:
-        # Reset the objective to the one used in the iteration, walk around
-        # the context manager for speed
-        model.solver.objective.set_linear_coefficients(
-            {model.variables.fva_old_objective: 0,
-             reaction.forward_variable: 1, reaction.reverse_variable: -1})
-        if all_fluxes:
-            current = loopless
-        return current
-
-    # If previous optimum was not in the loopless solution create a new
-    # almost loopless solution containing only loops including the current
-    # reaction. Than remove all of those loops.
     with model:
+        model.objective = 1.0 * model.variables.fva_old_objective
+        __add_cycle_free(model, sol.fluxes)
+        model.solver.optimize()
+        flux = reaction.flux
+
+        # If the previous optimum is maintained in the loopless solution it was
+        # loopless and we are done
+        if abs(flux - current) < zero_cutoff:
+            if solution:
+                return sol
+            return current
+
+        # If previous optimum was not in the loopless solution create a new
+        # almost loopless solution containing only loops including the current
+        # reaction. Than remove all of those loops.
+        ll_sol = get_solution(model).fluxes
+        bounds = reaction.bounds
         reaction.bounds = (current, current)
-        almost_loopless = loopless_solution(model)
+        model.solver.optimize()
+        almost_ll_sol = get_solution(model).fluxes
+        reaction.bounds = bounds
         # find the reactions with loops using the current reaction and remove
         # the loops
         for rxn in model.reactions:
-            if ((abs(loopless[rxn.id]) < zero_cutoff) and
-                    (abs(almost_loopless[rxn.id]) > zero_cutoff)):
-                rxn.bounds = (0, 0)
-
-        # Globally reset the objective to the one used in the FVA iteration
-        model.solver.objective.set_linear_coefficients(
-            {model.variables.fva_old_objective: 0,
-             reaction.forward_variable: 1, reaction.reverse_variable: -1})
-
-    solution = model.optimize(objective_sense=None)
-    if all_fluxes:
-        best = solution.fluxes
-    else:
-        best = reaction.flux
+            rid = rxn.id
+            if ((abs(ll_sol[rid]) < zero_cutoff) and
+                    (abs(almost_ll_sol[rid]) > zero_cutoff)):
+                rxn.bounds = max(0, rxn.lower_bound), min(0, rxn.upper_bound)
+
+        if solution:
+            solution = model.optimize(objective_sense=None)
+            best = solution.fluxes
```

Comment:
when I try, I never reach this statement but seems if I did I wouldn't get the requested whole solution?